Knoema.Client/Client.cs

```diff
@@ -117,12 +117,18 @@ private Task<T> ApiPost<T>(string path, object obj)
 
 		public Task<IEnumerable<Dataset>> ListDatasets(string source = null, string topic = null, string region = null)
 		{
-			var query = string.Empty
-				.AddUrlParam("source", source)
-				.AddUrlParam("topic", topic)
-				.AddUrlParam("region", region);
 
-			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
+			if (string.IsNullOrEmpty(source ?? topic ?? region))
+				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
+			else
+			{
+				var payload = new Dictionary<string, string>();
```

Comment:
Rahul, sorry i send mail by mistake.

I wanted to wrote that this issue should be fixed. It's good to pass null
everywhere. And it will not fail for community, because no it fails here
for main site and community code will pass this line:

if (Context.Community == null)
throw new InvalidOperationException("No valid community present to perform
this action.");



On Fri, Aug 21, 2015 at 5:47 PM, rpanday <notifications@github.com> wrote:

> In Knoema.Client/Client.cs
> <https://github.com/Knoema/knoema-csharp-driver/pull/15#discussion_r37630475>
> :
>
> >
> > -			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
> > +			if (string.IsNullOrEmpty(source ?? topic ?? region))
> > +				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
> > +			else
> > +			{
> > +				var payload = new Dictionary<string, string>();
>
> @polyakova <https://github.com/polyakova>
> we don't have it here though
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Knoema/knoema-csharp-driver/pull/15/files#r37630475>.
>



-- 
Evgeniya Polyakova,
Knoema